server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeInternalRPCServiceImpl.java

```diff
@@ -945,6 +949,27 @@ private PathPatternTree filterPathPatternTree(PathPatternTree patternTree, Strin
     return filteredPatternTree;
   }
 
+  @Override
+  public TSStatus setRegionStatus(TSetRegionStatusReq req) {
+    try {
+      if (req.getRegionId().getType() == TConsensusGroupType.DataRegion) {
+        CommonDescriptor.getInstance()
+            .getConfig()
+            .getDataRegionStatusMap()
+            .put(new DataRegionId(req.getRegionId().getId()), RegionStatus.parse(req.getStatus()));
+      } else {
+        CommonDescriptor.getInstance()
+            .getConfig()
+            .getSchemaRegionStatusMap()
+            .put(
+                new SchemaRegionId(req.getRegionId().getId()), RegionStatus.parse(req.getStatus()));
+      }
+      return new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+    } catch (Exception e) {
+      return new TSStatus(TSStatusCode.SET_REGION_STATUS_ERROR.getStatusCode());
```

Comment:
What scenario will trigger this error? It's better not to just use catch(Exception e) clause and don't write any error messages, which will cause confuse. For example, the most likely error is a failed to pares RegionStatus. Therefore, we should catch(IllegalException e) and set message "Unknown RegionStatus".